python/src/tstools/unrFiles.py

```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/env python3
+
+"""
+Module for downloading and handling UNR generated files.
+"""
+
+import requests as req
+import numpy as np
+
+from tstools import inputFileIO as ifio
+from tstools.util.convtime import convtime, yy_to_yyyy
+from tstools.util.nutils import msg_err
+
+########################################################################
+"""
+Constants
+"""
+
+ALL = 'all'
+BLANK_STR = '' 
+
+EQUIP_CHNG_CODE = 1
+EQ_CODE = 2
+
+JAN, FEB, MAR, APR, MAY = ['JAN','FEB','MAR','APR','MAY']
+JUN, JUL, AUG, SEP, OCT = ['JUN','JUL','AUG','SEP','OCT']
+NOV, DEC = ['NOV','DEC']
+
+########################################################################
+class UnrTxyz2:
+
+    """
+    Download, read in local copy, modify, or write out UNR-style .txyz2
+    position time series files.
+    """
+
+    ####################################################################
+    def __init__(self, site=''):
+
+        self.baseUrl = ('http://geodesy.unr.edu/gps_timeseries/txyz/'
+                       +'IGS08/')
+        if site != BLANK_STR:
+            site = site.upper()
+            self.site = site
+        else:
+            self.site = BLANK_STR
+        self.url = BLANK_STR
+        self.lineList = []
+
+    ####################################################################
+    def fetch(self, site='', baseUrl=''):
+
+        """
+        Read UNR .txyz2 position time series file from UNR website for
+        specified station.
+        """
+
+        if baseUrl != BLANK_STR:
+            self.baseUrl = baseUrl
+
+        if site != BLANK_STR:
+            site = site.upper()
+            self.site = site
```

Comment:
This is kind of a weird pattern to me. Normally you try to keep methods from having side affects that aren't obvious. For example, if I call `#fetch` with either `site` or `baseUrl` passed in, I'm modifying my `UnrTyz2` object, which I wouldn't expect if I was using this module.

I don't think its an issue necessarily, but worth adding to the cleanup TODO list.